src/grt/src/GlobalRouter.cpp

```diff
@@ -1523,7 +1521,7 @@ void GlobalRouter::addGuidesForLocalNets(odb::dbNet* db_net,
 {
   std::vector<Pin>& pins = db_net_map_[db_net]->getPins();
   int last_layer = -1;
-  for (uint p = 0; p < pins.size(); p++) {
+  for (int p = 0; p < pins.size(); p++) {
```

Comment:
Shouldn't this be size_t to avoid potentially overflowing the counter? Same question on the other instances of int in this change.

Reply:
It wouldn't hurt but a single net with >2^31 pins would be bizarre.

Reply:
Agreed. I'm sure in most cases it would be fine, but for correctness sake I think it's worth the conversion. A large number of these warning appear when compiling with clang, and I think it would just be good practice to avoid this class of error.